google-http-client/src/main/java/com/google/api/client/util/OpenCensusUtils.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.api.client.util;
+
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpStatusCodes;
+
+import io.opencensus.contrib.http.util.HttpPropagationUtil;
+import io.opencensus.trace.BlankSpan;
+import io.opencensus.trace.EndSpanOptions;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.Status;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.propagation.TextFormat;
+
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+/**
+ * Utilities for Census monitoring and tracing.
+ *
+ * @since 1.24
+ * @author Hailong Wen
+ */
+public class OpenCensusUtils {
+
+  private static final Logger LOGGER = Logger.getLogger(OpenCensusUtils.class.getName());
+
+  /**
+   * OpenCensus tracing component.
+   * When no OpenCensus implementation is provided, it will return a no-op tracer.
+   */
+  static Tracer tracer = Tracing.getTracer();
+
+  /**
+   * {@link TextFormat} used in tracing context propagation.
+   */
+  @Nullable
+  static TextFormat propagationTextFormat = null;
+
+  /**
+   * {@link TextFormat.Setter} for {@link activeTextFormat}.
+   */
+  @Nullable
+  static TextFormat.Setter propagationTextFormatSetter = null;
+
+  /**
+   * Sets the {@link TextFormat} used in context propagation.
+   * @param textFormat the text format.
+   */
+  public static void setPropagationTextFormat(@Nullable TextFormat textFormat) {
```

Comment:
Why is this public? Should be package protected and probably with an annotation that is for testing only.